packages/kotti-ui/source/kotti-field-file-upload/components/DropArea.vue

```diff
@@ -0,0 +1,265 @@
+		acceptLabel: { default: null, type: String },
+		allowMultiple: { default: true, type: Boolean },
+		dataTest: { default: null, type: String },
+		icon: { default: null, type: String },
+		infoUrl: { default: null, type: String },
+		isDisabled: { default: false, type: Boolean },
+		maxFileSize: { default: null, type: Number },
+		tabIndex: { default: 0, type: Number },
+	},
+	setup(
+		props: Pick<
+			KottiFieldFileUpload.PropsInternal,
+			| 'accept'
+			| 'acceptLabel'
+			| 'allowMultiple'
+			| 'dataTest'
+			| 'icon'
+			| 'infoUrl'
+			| 'isDisabled'
+			| 'maxFileSize'
+			| 'tabIndex'
+		>,
+		{ emit },
+	) {
+		const translations = useTranslationNamespace('KtFieldFileUpload')
+
+		const dropAreaRef = ref<HTMLDivElement | null>(null)
+		const isDragging = ref<boolean>(false)
+		const isHovering = ref<boolean>(false)
+		const uploadInputRef = ref<HTMLInputElement | null>(null)
+
+		const emitFiles = (files: FileList | null) => {
+			if (props.isDisabled || !files || files.length === 0) return
+
+			const { accept, allowMultiple, maxFileSize } = props
+			const validation = validateSelection({
+				accept,
+				allowMultiple,
+				fileName: files[0].name,
+				fileSize: files[0].size,
+				maxFileSize,
+				selection: files.length,
+			})
+
+			if (validation !== KottiFieldFileUpload.Validation.SUCCESS) {
+				emit('error', translations.value.validationMsg[validation])
+				return
+			}
+
+			emit('addFiles', Array.from(files).map(getFileItem))
+		}
+
+		const dropAreaListener = (event: KeyboardEvent) => {
+			if (
+				!props.isDisabled &&
+				[KEY_ENTER, KEY_SPACE].includes(event.key) &&
+				dropAreaRef.value === window.document.activeElement
+			) {
+				event.preventDefault()
+				event.stopPropagation()
+				uploadInputRef.value?.click()
+			}
+		}
+
+		onBeforeMount(() => window.addEventListener('keydown', dropAreaListener))
+
+		onUnmounted(() => window.removeEventListener('keydown', dropAreaListener))
+
+		return {
+			_icon: computed(() => props.icon ?? Yoco.Icon.CLOUD_UPLOAD),
+			_tabIndex: computed(() =>
+				props.isDisabled ? null : props.tabIndex ?? 0,
+			),
+			classes: computed(() => ({
+				'kt-field-file-upload__drop-area': true,
+				'kt-field-file-upload__drop-area--disabled': props.isDisabled,
+				'kt-field-file-upload__drop-area--hover':
+					!props.isDisabled && (isDragging.value || isHovering.value),
+			})),
+			dropAreaRef,
+			informationText: computed(() => {
+				if (!props.accept && !props.acceptLabel && !props.maxFileSize)
+					return null
+
+				return [
+					props.acceptLabel ?? formatAccept(props.accept),
+					props.maxFileSize
+						? `(${translations.value.text.max} ${formatFileSize(
+								props.maxFileSize,
+						  )})`
+						: null,
+				]
+					.filter((value) => Boolean(value))
+					.join(' ')
+			}),
+			isHovering,
+			isDragging,
+			onClick: () => {
+				if (!uploadInputRef.value || props.isDisabled) return
+				uploadInputRef.value.click()
```

Comment:
same question as 

what am I missing

the onClick programatically triggers a click on `uploadInputRef` which does nothing with the click 

https://github.com/3YOURMIND/kotti/pull/719/files#r1108635019

Reply:
This was due to the label which captures all clicks inside and then forwards the events to the first child inside. It's now refactored.